base.scad

```diff
@@ -1,9 +1,124 @@
+	render(){ // to get rid off the interference
+		translate([0,0,wall_thickness]) {
+			union() {
+				difference() {
+					cube([width+wall_thickness, width+wall_thickness, height]);
+					cube([width, standoff_width, height]);				
+				}
+				translate([width, width,0])
+					rotate([0,0,180])
+						intersection(){
+							cylinder(h = clearance, r = width,  center=false);
+							cube([width, width, height]);
+				}
+			}
+		}
+	}
 }
+
+// place 4 standoffs around a rectangular board space
+module standoffs(length, width, clearance) {
+	translate([(length/2) - standoff_width , (width/2) - standoff_width]) rotate([0,0,0])   single_standoff(standoff_height, standoff_width, clearance);
+	translate([(length/2) - standoff_width , -(width/2) + standoff_width]) rotate([0,0,270]) single_standoff(standoff_height, standoff_width, clearance);
+	translate([-(length/2) + standoff_width , -(width/2) + standoff_width]) rotate([0,0,180]) single_standoff(standoff_height, standoff_width, clearance);
+	translate([-(length/2) + standoff_width , (width/2) - standoff_width]) rotate([0,0,90])   single_standoff(standoff_height, standoff_width, clearance);
+}
+
+module connectors_female(angle) {
+	width = 10; // in degrees
+	rotate([0,0,angle-width]) 
+		rotate_extrude(angle=width, $fn=200) 
+			translate([-base_radius+wall_thickness ,base_height-2]) 
+				polygon( points=[[0,0],[0,5],[4,5],[4,2], [1,2],[1,3],[2,4],[3,4],[3,2]], paths=[[0,1,2,3], [4,5,6,7,8]] );
+}
+
+// cut 
+module venting_holes(xnum, ynum) {
+	width = 1;
+	spacing = 1.5;
+	for(y = [0 : width+spacing : (width+spacing)*ynum])
+	{
+		for(x = [0 : width+spacing : (width+spacing)*xnum])
+		{
+			translate([x - ((width+spacing)*xnum/2),-base_radius*1.25, y - ((width+spacing)*ynum/2) + base_height/2])
+			cube([width,2.5*base_radius,width]);
+		}
+	}
+}
+
+// cut a recess with port access into base
+// parameters are length/width of port access hole and 
+// zPos is height of port center over underside of PCB
+module port_access(length, height, zPos) {
+	difference(){
+		union(){	
+			children();	
+			// add new inner wall		
+			intersection(){
+				translate([0, 0, base_height/2])
+					    cylinder(h=base_height, d=width, center=true);		
+				translate([board_lenght/2, -(2*base_radius+5)/2 , 0]){ 
+					cube([base_radius - board_lenght/2 + 5, 2*base_radius+5, standoff_height + wall_thickness]);
+				}
+			}		
+		}		
+		// cut outer overhang
+		translate([board_lenght/2 + wall_thickness, -(2*base_radius+5)/2, -wall_thickness]){ 
+			cube([base_radius - board_lenght/2 + 5, 2*base_radius+5 , standoff_height + wall_thickness]);
+		}
+		// cut slope
+		translate([board_lenght/2 + wall_thickness, -(2*base_radius+5)/2, standoff_height])
+			rotate([0,60,0])
+				cube([base_radius - board_lenght/2 + 5, 2*base_radius+5 , standoff_height + wall_thickness]);
+		// cut port hole
+		translate([board_lenght/2 - 1, -height/2, -height/2 + wall_thickness + ground_clearance + zPos])
```

Comment:
I believe the second parameter of translate here should read `-length/2`, otherwise the port would be incorrectly centered? Also just a heads up, note that you got a typo in `board_lenght` (should be length).